internal/stores/metadata.go

```diff
@@ -69,13 +63,20 @@ type (
 		Host        dbHost
 		RenewedFrom fileContractID `gorm:"index;size:32"`
 
-		ProofHeight    uint64 `gorm:"index"`
-		RevisionHeight uint64 `gorm:"index"`
-		RevisionNumber string `gorm:"NOT NULL"` // string since db can't store math.MaxUint64
+		ContractMetadata
+		ContractSpending
```

Comment:
That was my first idea, but then I realised that `FCID` would have to be in `Common` as well and then we have to change queries like `Where({FCID: fcid})` to `Where(ContractCommon: ContractCommon{FCID: fcid})`... I also figured that having them separate might come in handy when we add more code that deals with a contract's financials.

I'll make the change though and see how it feels. I think `Host` is better kept out of it because there's no FK constraint between the archived contract and the host. So the types are different... I figured that was done on purpose so decided to leave it out.

I called it `ContractCommon` because calling it `dbContractCommon` would un-expose all fields, also it's more in line with `Model`.